src/tools/replay/nodes/TraceContextNode.java

```diff
@@ -2,11 +2,13 @@
 
 import com.oracle.truffle.api.dsl.Specialization;
 
-import som.primitives.processes.ChannelPrimitives.Process;
+import som.primitives.processes.ChannelPrimitives.TracingProcess;
+import som.primitives.threading.TaskThreads.TracedForkJoinTask;
+import som.primitives.threading.TaskThreads.TracedThreadTask;
 import som.vm.Activity;
 import tools.concurrency.TracingActors.TracingActor;
 import tools.replay.TraceRecord;
-import tools.replay.actors.ActorExecutionTrace.ActorTraceBuffer;
+import tools.replay.actors.UniformExecutionTrace.UniformTraceBuffer;
 
 
 public abstract class TraceContextNode extends TraceNode {
```

Comment:
This looks like something that could be simplified with a `ClassProfile` instead of the various specializations.
It seems we really just need the concrete type to resolve the virtual method calls, right?